src/targets/gpu/kernels/include/migraphx/kernels/bit_cast.hpp

```diff
@@ -23,15 +23,31 @@
 #define MIGRAPHX_GUARD_KERNELS_BITCAST_HPP
 
 #include <migraphx/kernels/type_traits.hpp>
+#include <migraphx/kernels/vec.hpp>
 
 namespace migraphx {
+
 template <typename To,
           typename From,
+          MIGRAPHX_REQUIRES(is_any_vec<To>()),
           MIGRAPHX_REQUIRES(is_trivially_copyable<To>{} and is_trivially_copyable<From>{})>
 inline constexpr To bit_cast(From fr) noexcept
+{
+    return vec_transform(fr)([](auto x) -> To {
+        static_assert(sizeof(To) == sizeof(decltype(x)));
+        return __builtin_bit_cast(To, x);
+    });
+}
+
+template <typename To,
+          typename From,
+          MIGRAPHX_REQUIRES(not is_any_vec<To>()),
+          MIGRAPHX_REQUIRES(is_trivially_copyable<To>{} and is_trivially_copyable<From>{})>
+inline constexpr auto bit_cast(From fr) noexcept
```

Comment:
There is no need for this overload, since the `vec_transform` works with non vectors as well.

Reply:
It doesn't work properly with the usage of bit_cast in `float8_impl`.

Reply:
We shouldn't be vectorizing fp8 types.

Reply:
Also, `vec_transform` already does the same check for `is_any_vec` and it just calls the function directly, so I dont see how that would cause an error with fp8 types.